packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -285,6 +291,21 @@ export class LoadTestDataStoreModel {
         }
     }
 
+    private deferUntilConnected(
+        callback:() => void,
+        errorHandler: (error) => void,
+    ) {
+        Promise.resolve().then(() => {
+            if (this.runtime.connected) {
```

Comment:
I'm not sure why "connected" part is here, i.e. the code before did not check for connected. More over, I think "lost" callbacks are running as result of disconnect.

Reply:
A bit of background info for this:

- initially, when the reentry check was enabled in the tests, these events which produced ops as a result on an op were scheduled in the next js turn. This caused some incidents as the container was closed when the callback was invoked.
- the next change checked for the runtime to be disposed, which also caused some incidents as the runtime was not disposed but the container was not connected (therefore publishing errors about submitting ops while not connected)
- I believe this approach is the right one as it both checks for the connection state and also ensures that if the container is temporarily disconnected, the op will be sent when it reconnects.

Reply:
That looks fine in terms of ensuring code that had to run connected will run connected.
It's very hard to say (for me) if that also causes any other cases (code that run previously on disconnect, like possibly "lost" handler) to run now connected, i.e. change in behavior that might not be expected here.
I simply do not have enough context (when certain events are fired - expectations on who and how handles connectivity transitions) to say that this change has no unwanted changes in behavior.

Reply:
Don't we _want_ those lost handlers to get run, rather than being dropped? If not, then rather than re-enqueuing the callback to fire on the "connected" event, we can just drop it if the runtime is not connected in the new turn. That seems to me to be the only other option.

Reply:
I do not know what semantics code has _before_ changes. I'd guess that if connection is lost, "lost" handler would fire and the intention of the code to run callback right away, in disconnected state.
Proposed changes might be correct, I'm just stating it's very hard to evaluate it without understanding current expectations of this code.

Reply:
That makes sense. In the past when I have written code that relies on being connected but forgot to check that it was connected, it has always been a mistake and it was appropriate to fix it by retrying with `on('connected')`. But I agree, in general, it isn't necessarily the right approach. If the code _fails_ when disconnected, it seems that it should either be explicitly not called in the first place (dropped), or rescheduled with `on('connected')` (lest it just error and get swallowed by an unhandled promise). So I guess the question here is, @andre4i, do the tests require that the `callback` is run? If so, then dropping it will _certainly_ cause a failure, whereas retrying it when connected will only _maybe_ cause an unexpected change.

Reply:
To give more context, I did manage to reproduce the connection related errors which fired the incidents last week. In this particular case is that the ops were scheduled to run when the container was still 'catching up'. I can debug this right now and all cases in which the connection event handler is invoked is when the container is connected and caught up. My major worry here was that the test would discard these ops and instead of actually submitting ops, it would silently not do much and we'd have a false sense of security wrt to the stress tests not erroring out.